crates/pop-cli/src/common/chain.rs

```diff
@@ -20,14 +24,38 @@ pub(crate) async fn configure(
 	input_message: &str,
 	default_input: &str,
 	url: &Option<Url>,
+	filter_fn: fn(&RPCNode) -> bool,
 	cli: &mut impl Cli,
 ) -> Result<Chain> {
 	// Resolve url.
 	let url = match url {
 		Some(url) => url.clone(),
 		None => {
-			// Prompt for url.
-			let url: String = cli.input(input_message).default_input(default_input).interact()?;
+			// Ask the user if they want to enter URL manually or select from a list of well-known
+			// endpoints.
+			let manual = cli
+				.confirm("Do you want to enter the node URL manually?")
+				.initial_value(false)
+				.interact()?;
+			let url = if manual {
+				// Prompt for manual URL input
+				cli.input(input_message).default_input(default_input).interact()?
+			} else {
+				// Select from available endpoints
+				let chains = extract_chain_endpoints().await?;
+				let mut prompt = cli.select("Select a chain (type to filter):");
+				for (pos, node) in chains.iter().enumerate() {
+					if filter_fn(node) {
+						prompt = prompt.item(pos, &node.name, "");
+					}
+				}
+				let selected = prompt.filter_mode().interact()?;
+				let providers = &chains[selected].providers;
+				providers
+					.choose(&mut rand::rng())
```

Comment:
We do something similar here for pseudo-random URL rotation: https://github.com/r0gue-io/pop-cli/blob/main/crates/pop-chains/src/deployer_providers.rs#L132
Using that approach could help us drop the rand dependency.